test-data/unit/check-literal.test

```diff
@@ -1240,3 +1240,81 @@ indirect.Literal()
 [builtins fixtures/isinstancelist.pyi]
 [out]
 
+
+--
+-- Other misc interactions
+--
+
+[case testLiteralMeetsWithCallablesInLists]
+from typing import List, Callable, Union
+from typing_extensions import Literal
+
+a: Callable[[Literal[1]], int]
+b: Callable[[Literal[2]], str]
+c: Callable[[int], str]
+d: Callable[[object], str]
+e: Callable[[Union[Literal[1], Literal[2]]], str]
+
+arr1 = [a, a]
+arr2 = [a, b]
+arr3 = [a, c]
+arr4 = [a, d]
+arr5 = [a, e]
+
+reveal_type(arr1)  # E: Revealed type is 'builtins.list[def (Literal[1]) -> builtins.int]'
+reveal_type(arr2)  # E: Revealed type is 'builtins.list[builtins.function*]'
+reveal_type(arr3)  # E: Revealed type is 'builtins.list[def (Literal[1]) -> builtins.object]'
+reveal_type(arr4)  # E: Revealed type is 'builtins.list[def (Literal[1]) -> builtins.object]'
+reveal_type(arr5)  # E: Revealed type is 'builtins.list[def (Literal[1]) -> builtins.object]'
+
+lit: Literal[1]
+reveal_type(arr1[0](lit))  # E: Revealed type is 'builtins.int'
+reveal_type(arr2[0](lit))  # E: Revealed type is 'Any' \
+                           # E: Cannot call function of unknown type
+reveal_type(arr3[0](lit))  # E: Revealed type is 'builtins.object'
+reveal_type(arr4[0](lit))  # E: Revealed type is 'builtins.object'
+reveal_type(arr5[0](lit))  # E: Revealed type is 'builtins.object'
+[builtins fixtures/list.pyi]
+[out]
+
+[case testLiteralMeetsWithTypeVars]
+from typing import TypeVar, Callable, Union
+from typing_extensions import Literal
+
+T = TypeVar('T')
+def unify(func: Callable[[T, T], None]) -> T: pass
+
+def f1(x: Literal[1], y: Literal[1]) -> None: pass
+def f2(x: Literal[1], y: Literal[2]) -> None: pass
+def f3(x: Literal[1], y: int) -> None: pass
+def f4(x: Literal[1], y: object) -> None: pass
+def f5(x: Literal[1], y: Union[Literal[1], Literal[2]]) -> None: pass
+
+reveal_type(unify(f1))  # E: Revealed type is 'Literal[1]'
+reveal_type(unify(f2))  # E: Revealed type is 'None'
+reveal_type(unify(f3))  # E: Revealed type is 'Literal[1]'
+reveal_type(unify(f4))  # E: Revealed type is 'Literal[1]'
+reveal_type(unify(f5))  # E: Revealed type is 'Literal[1]'
+[out]
```

Comment:
TBH, having both these tests look a bit redundant (especially that you have very good unit tests). I would combine them into a single tests and check meeting some type one way, and some types another way.

Reply:
I consolidated both tests into one and pruned out some probably-overkill `reveal_type(...)` calls. I think testing the types in both ways looks tidier though, so kept that.